ttkwidgets/notebook.py

```diff
@@ -0,0 +1,935 @@
+        if not self._tabmenu:
+            self._tab_list.grid_remove()
+        self._btn_left.grid(row=0, column=1, sticky='ns', pady=(0, 1))
+        self._canvas.grid(row=0, column=2, sticky='ew')
+        self._btn_right.grid(row=0, column=3, sticky='ns', pady=(0, 1))
+        self._body.grid(row=1, columnspan=4, sticky='ewns', padx=1, pady=1)
+
+        ttk.Frame(self, height=1,
+                  style='separator.TFrame').place(x=1, anchor='nw',
+                                                  rely=1, height=1,
+                                                  relwidth=1)
+
+        self._border_left = ttk.Frame(self, width=1, style='separator.TFrame')
+        self._border_right = ttk.Frame(self, width=1, style='separator.TFrame')
+        self._border_left.place(bordermode='outside', in_=self._body, x=-1, y=-2,
+                                width=1, height=self._body.winfo_reqheight() + 2, relheight=1)
+        self._border_right.place(bordermode='outside', in_=self._body, relx=1, y=-2,
+                                 width=1, height=self._body.winfo_reqheight() + 2, relheight=1)
+
+        # --- bindings
+        self._tab_frame.bind('<Configure>', self._on_configure)
+        self._canvas.bind('<Configure>', self._on_configure)
+        self.bind_all('<ButtonRelease-1>', self._on_click)
+
+        self.config = self.configure
+        Notebook._initialized = True
+
+    def __getitem__(self, key):
+        return self.cget(key)
+
+    def __setitem__(self, key, value):
+        self.configure(**{key: value})
+    
+    def setup_style(self, bg="#dddddd", activebg="#efefef", pressedbg="#c1c1c1",
+                    fg="black", fieldbg="white", lightcolor="#ededed", darkcolor="##cfcdc8",
+                    bordercolor="#888888", focusbordercolor="#5e5e5e", selectbg="#c1c1c1",
+                    selectfg="black", unselectfg="#999999", disabledfg='#999999', disabledbg="#dddddd"):
+        """
+        Setups the style for the notebook.
+        :param bg: 
+        :param activebg: 
+        :param pressedbg: 
+        :param fg:
+        :param fieldbg:
+        :param lightcolor:
+        :param darkcolor:
+        :param bordercolor:
+        :param focusbordercolor:
+        :param selectbg:
+        :param selectfb:
+        :param unselectfg:
+        :param disabledfg:
+        :param disabledbg:
+        """
+        theme = {'bg': bg,
+                 'activebg': activebg,
+                 'pressedbg': pressedbg,
+                 'fg': fg,
+                 'fieldbg': fieldbg,
+                 'lightcolor': lightcolor,
+                 'darkcolor': darkcolor,
+                 'bordercolor': bordercolor,
+                 'focusbordercolor': focusbordercolor,
+                 'selectbg': selectbg,
+                 'selectfg': selectfg,
+                 'unselectedfg': unselectfg,
+                 'disabledfg': disabledfg,
+                 'disabledbg': disabledbg}
+
+        self.images = (
```

Comment:
Instance attribute defined outside `__init__`. The existence of this attribute should be defined in `__init__`, though it may be initialized as an empty tuple.